From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: tabrisnet <tabris@tabris.net>
Date: Sun, 12 Oct 2025 15:21:20 +0000
Subject: mt7988a bpi-r4 mt7988a-bananapi-bpi-r4-sd.dtb

OpenWRT uses some extra machinery for producing flattened DTBs that
aren't in the kernel. We don't support that, so instead this patches
the kernel tree.

Signed-off-by: tabrisnet <tabris@tabris.net>
---
 arch/arm64/boot/dts/mediatek/Makefile | 12 ++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/Makefile b/arch/arm64/boot/dts/mediatek/Makefile
index d728953c10ad..15041f497529 100644
--- a/arch/arm64/boot/dts/mediatek/Makefile
+++ b/arch/arm64/boot/dts/mediatek/Makefile
@@ -19,14 +19,24 @@ dtb-$(CONFIG_ARCH_MEDIATEK) += mt7986a-bananapi-bpi-r3-nand.dtbo
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7986a-bananapi-bpi-r3-nor.dtbo
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7986a-bananapi-bpi-r3-sata.dtbo
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7986a-bananapi-bpi-r3-sd.dtbo
 #dtb-$(CONFIG_ARCH_MEDIATEK) += mt7986a-rfb.dtb
 #dtb-$(CONFIG_ARCH_MEDIATEK) += mt7986b-rfb.dtb
+
+mt7988a-bananapi-bpi-r4-sd-dtbs := \
+	mt7988a-bananapi-bpi-r4.dtb \
+	mt7988a-bananapi-bpi-r4-sd.dtbo
+mt7988a-bananapi-bpi-r4-emmc-dtbs := \
+	mt7988a-bananapi-bpi-r4.dtb \
+	mt7988a-bananapi-bpi-r4-emmc.dtbo
+
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-2g5.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-emmc.dtbo
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-sd.dtbo
+dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-emmc.dtb
+dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-sd.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt8167-pumpkin.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt8173-elm.dtb
 #dtb-$(CONFIG_ARCH_MEDIATEK) += mt8173-elm-hana.dtb
 #dtb-$(CONFIG_ARCH_MEDIATEK) += mt8173-elm-hana-rev7.dtb
 #dtb-$(CONFIG_ARCH_MEDIATEK) += mt8173-evb.dtb
@@ -108,7 +118,9 @@ dtb-$(CONFIG_ARCH_MEDIATEK) += mt8183-evb.dtb
 
 # Device tree overlays support
 DTC_FLAGS_mt7986a-bananapi-bpi-r3 := -@
 DTC_FLAGS_mt7986a-bananapi-bpi-r3-mini := -@
 DTC_FLAGS_mt7988a-bananapi-bpi-r4 := -@
+DTC_FLAGS_mt7988a-bananapi-bpi-r4-emmc := -@
+DTC_FLAGS_mt7988a-bananapi-bpi-r4-sd := -@
 DTC_FLAGS_mt7988a-bananapi-bpi-r4-2g5 := -@
 DTC_FLAGS_mt8395-radxa-nio-12l := -@
-- 
Created with Armbian build tools https://github.com/armbian/build
